focus-within.js

```diff
@@ -118,13 +118,25 @@ export const focusWithin = (baseElement) => class extends baseElement {
 
   /**
    * Set `_focusWithin` as a `true`.
+   * When focused element is removed, triggers `_removeFocus`.
    * @protected
    */
-  _setFocusWithin() {
+  _setFocusWithin(e) {
     if (this._focusoutTimeoutId) {
       clearTimeout(this._focusoutTimeoutId);
     }
     this._focusWithin = true;
+    const target = e && e.composedPath() && e.composedPath()[0];
+    if (target) {
+      let intervalId = setInterval(() => {
```

Comment:
@nirmalbaldaniya-dream This timer should get cancelled when focus is removed. 

At present, it's cancelled only if element is detached. Same way, test another case where focus is shifted from one of the inner element to another inner element.